test/e2e/testnet/manifest.go

```diff
@@ -0,0 +1,77 @@
+package testnet
+
+import (
+	"time"
+
+	"github.com/celestiaorg/celestia-app/v2/app"
+	"github.com/celestiaorg/celestia-app/v2/app/encoding"
+	"github.com/celestiaorg/celestia-app/v2/test/util/genesis"
+	blobtypes "github.com/celestiaorg/celestia-app/v2/x/blob/types"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+)
+
+// Manifest defines the parameters for a testnet.
+type Manifest struct {
+	ChainID      string
+	TestDuration time.Duration
+	// Number of validators in the testnet
+	Validators int
+	TxClients  int
+	// Self-delegation amount for validators
+	SelfDelegation int64
+	// CelestiaAppVersion a specific version of the celestia-app container image within celestiaorg repository on GitHub's Container Registry i.e., https://github.com/celestiaorg/celestia-app/pkgs/container/celestia-app
+	CelestiaAppVersion string
+	// TxClientVersion  a specific version of the txsim container image within celestiaorg repository on GitHub's Container Registry, i.e., https://github.com/celestiaorg/celestia-app/pkgs/container/txsim
+	TxClientVersion string
+	// Resource requirements for a validator node
+	ValidatorResource Resources
+	// Resource requirements for a tx client
+	TxClientsResource Resources
+
+	// tx client settings
+	// Number of blobs per sequence
+	BlobsPerSeq int
+	// Number of blob sequences
+	BlobSequences int
+	// Size of blobs in bytes, e.g., "10000" (exact size) or "10000-20000" (min-max format)
+	BlobSizes string
+
+	// p2p configs
+	// Bandwidth per peer in bytes per second
+	PerPeerBandwidth int64
+	// consensus configs
+	TimeoutCommit  time.Duration
+	TimeoutPropose time.Duration
+
+	// Mempool configs
+	// Mempool version
+	Mempool      string
+	BroadcastTxs bool
+
+	// prometheus configs
+	Prometheus bool
+
+	// consensus parameters
+	MaxBlockBytes int64
+
+	// other configs
+	UpgradeHeight    int64
+	GovMaxSquareSize int64
+}
+
+func (tm *Manifest) GetGenesisModifiers() []genesis.Modifier {
```

Comment:
What's the rationale for `tm` instead of `m`?

Reply:
It was a remnant from an earlier state of the code where the `Manifest` was initially `TestManifest`, thus the function receiver name was `tm`. Good catch, fixed it now: https://github.com/celestiaorg/celestia-app/commit/5eea75475272900b57d9a1cba565c65953d0f4ee.